connectors/byoc.py

```diff
@@ -498,11 +498,27 @@ async def reset_sync_now_flag(self):
 
         script = {
             "lang": "painless",
+            # only reset sync_now flag if it's set, otherwise set op to 'noop' (the returned 'result' will be 'noop')
             "source": "if (ctx._source.sync_now) { ctx._source.sync_now = false } else { ctx.op = 'noop' }",
         }
         resp = await self.index.update_by_script(doc_id=self.id, script=script)
         return resp["result"] == "updated"
 
+    async def update_last_sync_scheduled_at(self, new_ts):
+        """Update last_sync_scheduled_at with update_by_script, and return True only if last_sync_scheduled_at is
+        updated."""
+        script = {
+            "lang": "painless",
+            # only update last_sync_scheduled_at to new_ts if it's not updated by other instances, otherwise set op to 'noop' (the returned 'result' will be 'noop')
+            "source": "if (ctx._source.last_sync_scheduled_at == params['old_ts']) { ctx._source.last_sync_scheduled_at = params['new_ts'] } else { ctx.op = 'noop' }",
```

Comment:
if we use painless, can we have the script as a variable at the top to make it readable?

something like:

```
_UPDATE_SOURCE = """\
if (ctx._source.last_sync_scheduled_at == params['old_ts']) { 
    ctx._source.last_sync_scheduled_at = params['new_ts'] 
} else { 
    ctx.op = 'noop' 
}
"""
```